pkgs/development/libraries/spdlog/default.nix

```diff
@@ -34,12 +34,18 @@ stdenv.mkDerivation rec {
   propagatedBuildInputs = [ fmt ];
   checkInputs = [ catch2_3 ];
 
+  # Don't build tests if not needed
+  preConfigure = ''
+    if [ -n "$doCheck" ]; then
+      cmakeFlagsArray+=("-DSPDLOG_BUILD_TESTS=ON")
+    fi
+  '';
+
   cmakeFlags = [
     "-DSPDLOG_BUILD_SHARED=${if staticBuild then "OFF" else "ON"}"
     "-DSPDLOG_BUILD_STATIC=${if staticBuild then "ON" else "OFF"}"
     "-DSPDLOG_BUILD_EXAMPLE=OFF"
     "-DSPDLOG_BUILD_BENCH=OFF"
```

Comment:
```suggestion
    "-DSPDLOG_BUILD_BENCH=OFF"
    "-DSPDLOG_BUILD_TESTS=${if doCheck && stdenv.hostPlatform == stdenv.targetPlatform then "ON" else "OFF"}"
```

Reply:
Use `finalAttrs.finalPackage.doCheck` to reuse the mkDerivation logic

Reply:
TIL about `finalAttrs`.